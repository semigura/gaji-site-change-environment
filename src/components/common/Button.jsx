import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'gatsby';
import Icon from './icons/Icon';

import css from './Button.module.scss';

const Button = props => {
  const { className, isSmall, label, onClickButton, to, withIcon } = props;
  const classes = classNames(css.Button, { [css.isSmall]: isSmall }, className);

  return to ? (
    <Link to={to} className={classes}>
      {label}
      {withIcon && (
        <Icon
          className={css.Button__Icon}
          name="arrow-right"
          aria-hidden="true"
        />
      )}
    </Link>
  ) : (
    <button type="button" className={classes} onClick={onClickButton}>
      {label}
      {withIcon && (
        <Icon
          className={css.Button__Icon}
          name="arrow-right"
          aria-hidden="true"
        />
      )}
    </button>
  );
};

Button.propTypes = {
  className: PropTypes.string,
  isSmall: PropTypes.bool,
  label: PropTypes.string.isRequired,
  onClickButton: PropTypes.func,
  to: PropTypes.string,
  withIcon: PropTypes.bool,
};

Button.defaultProps = {
  className: undefined,
  isSmall: false,
  onClickButton: undefined,
  to: undefined,
  withIcon: true,
};

export default Button;
