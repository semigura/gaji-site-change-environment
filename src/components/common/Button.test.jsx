import { shallow } from 'enzyme';
import React from 'react';

import Button from './Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Button label="button label" />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('with isSmall', () => {
    it('should attach class', () => {
      const wrapper = shallow(<Button label="button label" isSmall />);
      expect(wrapper.find('.Button').hasClass('isSmall')).toBeTruthy();
    });
  });
});
