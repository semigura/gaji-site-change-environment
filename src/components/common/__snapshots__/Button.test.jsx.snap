// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="Button"
  type="button"
>
  button label
  <Icon
    aria-hidden="true"
    className="Button__Icon"
    name="arrow-right"
  />
</button>
`;
